dace/transformation/dataflow/otf_map_fusion.py

```diff
@@ -96,102 +144,185 @@ def apply(self, graph: SDFGState, sdfg: SDFG):
 
             intermediate_dnodes.add(node)
 
-        # Remove edges of these nodes to second map
+        # b. Remove edges of these nodes to second map
         for dnode in intermediate_dnodes:
             for edge in graph.edges_between(dnode, self.second_map_entry):
                 graph.remove_edge_and_connectors(edge)
 
-        # Add edges for input of first map to second map
+        # c. Add edges for in access nodes of first map to second map
         connector_mapping = {}
         for edge in graph.in_edges(first_map_entry):
             old_in_connector = edge.dst_conn
-            new_in_connector = self.second_map_entry.next_connector(old_in_connector)
-            if self.second_map_entry.add_in_connector(new_in_connector):
-                memlet = copy.deepcopy(edge.data)
-                graph.add_edge(edge.src, edge.src_conn, self.second_map_entry, new_in_connector, memlet)
-
-                connector_mapping[old_in_connector] = new_in_connector
+            if not edge.data is None and not edge.data.data is None:
+                suffix = edge.data.data
             else:
-                raise ValueError("Failed to connect")
+                suffix = "OTF"
+
+            new_in_connector = "IN_" + suffix
+            if not self.second_map_entry.add_in_connector(new_in_connector):
+                for n in range(2, 65536):
+                    new_in_connector = "IN_" + suffix + "_" + str(n)
+                    if self.second_map_entry.add_in_connector(new_in_connector):
+                        break
```

Comment:
I still don't understand why you do not use `next_connector`

Reply:
Yeah, super annoying. The solution is awkward, but calling "while add_in_connector == False: new_connector = next_connector(new_connector)" was causing problems. I couldn't debug them, but can try another time. I know that it occurs in hdiff when fusing all maps

Reply:
I will look into this separately